analyzers/tests/SonarAnalyzer.UnitTest/Rules/AssertionsShouldBeCompleteTest.cs

```diff
@@ -0,0 +1,109 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.Rules.CSharp;
+using SonarAnalyzer.UnitTest.MetadataReferences;
+
+namespace SonarAnalyzer.UnitTest.Rules;
+
+[TestClass]
+public class AssertionsShouldBeCompleteTest
+{
+    private readonly VerifierBuilder fluentAssertions = new VerifierBuilder<AssertionsShouldBeComplete>()
+        .AddReferences(NuGetMetadataReference.FluentAssertions("6.10.0"))
+        .AddReferences(MetadataReferenceFacade.SystemXml)
+        .AddReferences(MetadataReferenceFacade.SystemXmlLinq)
+        .AddReferences(MetadataReferenceFacade.SystemNetHttp)
+        .AddReferences(MetadataReferenceFacade.SystemData);
+
+    private readonly VerifierBuilder nfluent = new VerifierBuilder<AssertionsShouldBeComplete>()
+        .AddReferences(NuGetMetadataReference.NFluent("2.8.0"));
+
+    private readonly VerifierBuilder nsubstitute = new VerifierBuilder<AssertionsShouldBeComplete>()
+        .AddReferences(NuGetMetadataReference.NSubstitute("5.0.0"));
+
+    private readonly VerifierBuilder allFrameworks;
+
+    public AssertionsShouldBeCompleteTest()
+    {
+        allFrameworks = new VerifierBuilder<AssertionsShouldBeComplete>()
+            .AddReferences(fluentAssertions.References)
+            .AddReferences(nfluent.References)
+            .AddReferences(nsubstitute.References);
+    }
+
+    [TestMethod]
+    public void AssertionsShouldBeComplete_FluentAssertions_CSharp7() =>
+        fluentAssertions
+        .WithOptions(ParseOptionsHelper.OnlyCSharp7)
+        .AddPaths("AssertionsShouldBeComplete.FluentAssertions.CSharp7.cs")
+        .Verify();
+
+    [TestMethod]
+    public void AssertionsShouldBeComplete_FluentAssertions_MissingParen() =>
+        fluentAssertions
+        .AddSnippet("""
+            using FluentAssertions;
+            public class Test
+            {
+                public void MissingParen()
+                {
+                    var s = "Test";
+                    s.Should(;  // Error
+                }
+            }
+            """)
+        .Verify();
```

Comment:
I had it there and extracted it. This is a syntax compiler error and we need to rely on the compiler to recover from the missing tokens. For that reason, I'd like to have this isolated from the rest of the tests.

Reply:
We have another compiler error in ...CSharp7.cs, which is why I thought, it might make sense to keep the other one in there as well. But generally speaking, I have no problem with separating test cases of course.

Reply:
These were binding errors that should not affect other code parts. Syntax errors can bring the syntax tree into an unknown state and may affect the code below.